### PR TITLE
ARROW-10941: [Doc] Document supported Parquet encryption features

### PR DESCRIPTION
**Pull request by ggershinsky**

### CODE REVIEW

**Comment by ggershinsky** [comment]
@thamht4190 Can you please have a look at this? To make sure I haven't missed anything.

**Comment by thamha-emotiv** [comment]
LGTM. Thanks!

**Comment by thamht4190** [comment]
LGTM. Thanks!

**Comment by ggershinsky** [comment]
@thamht4190 thanks. @pitrou - ready for your review.

File: docs/source/cpp/parquet.rst [hunk 1]
```diff
@@ -244,7 +244,22 @@ Limitations
...
+----------
+
+Parquet C++ implements all features, specified in the encryption `specification <https://github.com/apache/parquet-format/blob/master/Encryption.md>`__,
+except for encryption of column index and bloom filter modules. 
+
+More specifically, Parquet C++ supports
+
+* AES_GCM_V1 and AES_GCM_CTR_V1 encryption algorithms
+
+* AAD suffix for Footer, ColumnMetaData, Data Page, Dictionary Page, Data PageHeader, Dictionary PageHeader module types. ColumnIndex, OffsetIndex, BloomFilter Header, BloomFilter Bitset are not supported.
```
Comment (kiszk) [inline docs/source/cpp/parquet.rst:257]: nit: types. -> types,  ?

**Comment by pitrou** [comment]
I've polished this a bit, rebased, and added a sidebar to navigate more easily inside the documentation page. WIll merge, thank you @ggershinsky !

**Pull request closed without merging**